planner/core/rule_predicate_push_down.go

```diff
@@ -391,7 +391,7 @@ func (p *LogicalProjection) PredicatePushDown(predicates []expression.Expression
 	}
 	for _, cond := range predicates {
 		newFilter := expression.ColumnSubstitute(cond, p.Schema(), p.Exprs)
-		if !expression.HasGetSetVarFunc(newFilter) {
+		if !expression.HasGetSetVarFunc(newFilter) && expression.ExprFromSchema(newFilter, p.Children()[0].Schema()) {
```

Comment:
Will this cause regression? For example, 't2.b > 'a'' in #28743 cannot be pushed down to TiKV.